lib/core/src/sync/model/mod.rs

```diff
@@ -0,0 +1,50 @@
+use rusqlite::{
+    types::{FromSql, FromSqlError, FromSqlResult, ToSqlOutput, ValueRef},
+    ToSql,
+};
+
+pub(crate) mod sync;
+
+#[derive(Copy, Clone)]
+pub(crate) enum RecordType {
+    Receive = 0,
+    Send = 1,
+    Chain = 2,
+}
+
+impl ToSql for RecordType {
+    fn to_sql(&self) -> rusqlite::Result<ToSqlOutput<'_>> {
+        Ok(rusqlite::types::ToSqlOutput::from(*self as i8))
+    }
+}
+
+impl FromSql for RecordType {
+    fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
+        match value {
+            ValueRef::Integer(i) => match i as u8 {
+                0 => Ok(Self::Receive),
+                1 => Ok(Self::Send),
+                2 => Ok(Self::Chain),
+                _ => Err(FromSqlError::OutOfRange(i)),
+            },
+            _ => Err(FromSqlError::InvalidType),
+        }
+    }
+}
+
+pub(crate) struct SyncState {
+    pub(crate) data_id: String,
+    pub(crate) record_id: String,
+    pub(crate) record_revision: u64,
+}
+
+pub(crate) struct SyncSettings {
+    pub(crate) remote_url: String,
+    pub(crate) latest_revision: u64,
+}
+
+pub(crate) struct SyncOutgoingDetails {
+    pub(crate) record_id: String,
+    pub(crate) record_type: RecordType,
+    pub(crate) updated_fields_json: Option<String>,
```

Comment:
If this intends to contain the array of keys that were modified perhaps it is better to use the parsed structure? e.g slice of strings?

Reply:
We also need to keep a flag is_local somewhere so we know not to claim/refund cooperatively if we didn't initiate the swap.